internal/api/handler/achievement_handler.go

```diff
@@ -138,3 +134,35 @@ func ListAchievementsHandler(ctx context.Context, achSvc service.Service) func(h
 		})
 	}
 }
+
+// DeleteAchievementHandler returns an HTTP handler that deletes particular achievement using profileSvc.
+func DeleteAchievementHandler(ctx context.Context, achSvc service.Service) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		profileID, achID, err := helpers.GetMultipleParams(r)
+		if err != nil {
+			middleware.ErrorResponse(w, http.StatusBadGateway, err)
+			zap.S().Error("error while getting the IDs from request")
+			return
+		}
+		// Decode into the struct
+		req := decodeDeleteAchievementRequest(profileID, achID)
+
+		// call the service
+		err = achSvc.DeleteAchievement(ctx, req)
+		if err != nil {
+			if err == errors.ErrNoData {
+				middleware.SuccessResponse(w, http.StatusOK, specs.MessageResponse{
+					Message: "No data found for deletion",
+				})
+				return
+			}
+			middleware.ErrorResponse(w, http.StatusBadGateway, errors.ErrFailedToDelete)
+			zap.S().Error("error while deleting the achievements: ", err)
+			return
+		}
+
+		middleware.SuccessResponse(w, http.StatusOK, specs.MessageResponse{
+			Message: "Achievement deleted successfully",
+		})
+	}
```

Comment:
Add log here with profile ID and achievement ID

Reply:
done